### PR TITLE
new: dev: #223: add integration test for verifying cache eviction behavior

### PR DESCRIPTION
**Pull request by jordanw-bq**

### CODE REVIEW

File: rubix-tests/src/test/robotframework/acceptance/CacheEviction.robot [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+${METRIC_EVICTION}  rubix.bookkeeper.count.cache_eviction
+
+${REMOTE_PATH}      ${CURDIR}${/}rubixIntegrationTestFile
+${FILE_LENGTH}      1048576
+${LAST_MODIFIED}    1514764800
+${START_BLOCK}      0
+${END_BLOCK}        1048576
+${CLUSTER_TYPE}     3   # TEST_CLUSTER_MANAGER
+
+${CACHE_DIR_PFX}    /Users/jordanw/RobotTest/dtest/
```
Comment (abhishekdas99) [inline rubix-tests/src/test/robotframework/acceptance/CacheEviction.robot]: This needs to be some system level directory. The same way we create directories in our unit tests

File: rubix-tests/src/test/robotframework/acceptance/bks.robot [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Read test file data using FS
+    [Arguments]     ${fileName}  ${startBlock}  ${endBlock}
+    ${didRead} =    Read data using file system
+    ...             ${fileName}
+    ...             ${startBlock}
+    ...             ${endBlock}
+    Should be true  ${didRead}
+
+Verify cache directories
```
Comment (abhishekdas99) [inline rubix-tests/src/test/robotframework/acceptance/bks.robot]: Only the cache directory size is being verified. So change the method name to mention that to reduce ambiguity

File: rubix-tests/src/test/robotframework/acceptance/bks.robot [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    \   Append To List  ${optionsList}  -D${key}\=${value}
+    ${optionArgs} =     Join Command Line   ${optionsList}
+    [Return]            ${optionArgs}
+
+Generate test files
+    [Arguments]     ${numFiles}
+    ...             ${fileLength}
+    @{testFileList} =   Create List
+    :FOR    ${index}    IN RANGE    ${numFiles}
+    \   ${testFile} =        Set variable  ${REMOTE_PATH}${index}
```
Comment (abhishekdas99) [inline rubix-tests/src/test/robotframework/acceptance/bks.robot]: Where are we deleting the test file after the test is finished ?

File: rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java [hunk 4]
```diff
@@ -85,6 +85,7 @@
   private static final String KEY_ENABLE_FILE_STALESSNESS_CHECK = "rubix.enable.file.staleness-check";
   private static final String KEY_STALE_FILEINFO_EXPIRY_PERIOD = "rubix.stale.fileinfo.expiry.period";
   private static final String KEY_CLEANUP_FILES_DURING_START = "rubix.cleanup.files.during.start";
+  private static final String KEY_MAX_CACHE_SIZE = "hadoop.cache.data.fullness.size";
```
Comment (abhishekdas99) [inline rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java]: use rubix

File: rubix-tests/src/test/robotframework/acceptance/CacheEviction.robot (hunk 1 above)
Comment (jordanw-bq) [reply rubix-tests/src/test/robotframework/acceptance/CacheEviction.robot]: Updated to use OS's temporary directory for caching & holding data files

File: rubix-tests/src/test/robotframework/acceptance/bks.robot (hunk 2 above)
Comment (jordanw-bq) [reply rubix-tests/src/test/robotframework/acceptance/bks.robot]: Changed keyword name for clarification

File: rubix-tests/src/test/robotframework/acceptance/bks.robot (hunk 3 above)
Comment (jordanw-bq) [reply rubix-tests/src/test/robotframework/acceptance/bks.robot]: Files are now deleted as part of test teardown (line 75 below)
https://github.com/qubole/rubix/blob/ae9f82a296250d5ff92c399afab1c3e7b4b14d35/rubix-tests/src/test/robotframework/acceptance/CacheEviction.robot#L73-L75

File: rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java (hunk 4 above)
Comment (jordanw-bq) [reply rubix-spi/src/main/java/com/qubole/rubix/spi/CacheConfig.java]: Updated config option to use `rubix` namespace

File: rubix-client/pom.xml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                            <goal>jar</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>copy-for-rf</id>
+                        <goals>
+                            <goal>test-jar</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>../rubix-tests/src/test/resources/robotframework/libraries</outputDirectory>
```
Comment (abhishekdas99) [inline rubix-client/pom.xml]: Can we change the path for rubix-client-test jar from rubix-tests to rubix-client and change the necessary code where this is accessed.

https://github.com/qubole/rubix/pull/230/files#diff-840acf36052c63edee747f8906e91044R115

Comment (jordanw-bq) [reply rubix-client/pom.xml]: Removed JAR relocation in `rubix-client` and updated reference to client JAR in `rubix-tests`

**Pull request merged**